docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This may be realized as policies defined by providers to control a service
+in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is 
+ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan Components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
+
+#### Storage
+APIMan is capable of using several backends (e.g. ElasticSearch, MySql) to store configuration, policy, and metrics.  The initial integration will utilize ElasticSearch with the goal of reusing the same ElasticSearch cluster that is deployed for aggregated logging.  It is assumed when the backend resides on the cluster, [Origin Aggregated logging](https://github.com/openshift/origin-aggregated-logging)<sup>[4](#r4)</sup> will be deployed before APIMan. 
```

Comment:
This is the best argument made so far against reusing the cluster.

ES for logging is != the configuration for ES for anything else.

On Mon, Mar 21, 2016 at 12:25 PM, Luke Meyer <notifications@github.com>
wrote:

> In docs/proposals/apiman-integration.md
> <https://github.com/openshift/origin/pull/7207#discussion_r56851640>:
>
> > +
> > +## Motivation
> > +Providers of services may have the need to control who and how their services are consumed.  This may be realized as policies defined by providers to control a service
> > +in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>.
> > +
> > +## Constraints and Assumptions
> > +### Authentication
> > +The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is
> > +ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
> > +
> > +### Deployment
> > +#### Communication
> > +APIMan Components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
> > +
> > +#### Storage
> > +APIMan is capable of using several backends (e.g. ElasticSearch, MySql) to store configuration, policy, and metrics.  The initial integration will utilize ElasticSearch with the goal of reusing the same ElasticSearch cluster that is deployed for aggregated logging.  It is assumed when the backend resides on the cluster, [Origin Aggregated logging](https://github.com/openshift/origin-aggregated-logging)<sup>[4](#r4)</sup> will be deployed before APIMan.
>
> Tying these two unrelated services together by storage just doesn't smell
> right. ES is not just dumb storage for logging, it has very
> logging-specific plugins and configuration for access control. The use
> cases are totally different in terms of volume; imagine needing to do an ES
> data migration that takes hours/days due to logging data and also takes
> APIMan offline for that period. It feels like logging storage will be an
> albatross for APIMan. I don't think it's worth it for saving a little RAM
> for OS admins (I question whether in the end it saves on complexity at
> all). It doesn't need to be ES, could be postgres or mysql (though we would
> want to cluster them which could be a challenge). It could also just be a
> "small" ES cluster (ES does not have to be a monster if it doesn't have a
> lot of data).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7207/files/f85d81b38ace6bee04bb6b1826ab37022f012803#r56851640>
>